DevCycleTests/Models/DVCClientTest.swift

```diff
@@ -170,7 +172,67 @@ class DVCClientTest: XCTestCase {
         client.refetchConfig(sse: true, lastModified: 789)
         XCTAssertEqual(service.numberOfConfigCalls, 6)
     }
+    
+    func testSseCloseGetsCalledWhenBackgrounded() {
+        let service = MockService()
+        let user1 = try! DVCUser.builder().userId("user1").build()
+        let client = try! DVCClient.builder().user(user1).environmentKey("my_env_key").build(onInitialized: nil)
+        client.setup(service: service)
+        client.initialized = true
+        
+        let mockSSEConnection = MockSSEConnection()
+        client.sseConnection = mockSSEConnection
```

Comment:
if you're going to be assigning it like this, maybe it should use dependency injection instead? then we'd also be able to write other tests that rely on the sse connection receiving a message etc.

Reply:
i feel like this is cleaner, this is only possible through adding `@testable` in the test so it's not able to be set by actual users of the SDK.

Reply:
handling it this way going forward is going to be less cumbersome as we won't have to pass in different mocked classes